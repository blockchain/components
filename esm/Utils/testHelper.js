import React from 'react';
import { shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import theme from '../Tools/theme';

function wrapWithTheme(fn, children, options) {
  var wrapper = fn(React.createElement(ThemeProvider, {
    theme: theme
  }, children), options);
  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext()
  });
}

export function shallowWithTheme() {
  return wrapWithTheme.apply(void 0, [shallow].concat(Array.prototype.slice.call(arguments)));
}
export function mountWithTheme() {
  return wrapWithTheme.apply(void 0, [mount].concat(Array.prototype.slice.call(arguments)));
}