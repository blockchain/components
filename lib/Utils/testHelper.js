"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.shallowWithTheme = shallowWithTheme;
exports.mountWithTheme = mountWithTheme;

var _react = _interopRequireDefault(require("react"));

var _enzyme = require("enzyme");

var _styledComponents = require("styled-components");

var _theme = _interopRequireDefault(require("../Tools/theme"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function wrapWithTheme(fn, children, options) {
  var wrapper = fn(_react.default.createElement(_styledComponents.ThemeProvider, {
    theme: _theme.default
  }, children), options);
  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext()
  });
}

function shallowWithTheme() {
  return wrapWithTheme.apply(void 0, [_enzyme.shallow].concat(Array.prototype.slice.call(arguments)));
}

function mountWithTheme() {
  return wrapWithTheme.apply(void 0, [_enzyme.mount].concat(Array.prototype.slice.call(arguments)));
}