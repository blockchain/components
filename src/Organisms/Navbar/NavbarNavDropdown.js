import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const Wrapper = styled.li`
  position: relative;
  width: 100%;
  padding: 0.5rem 0;
  border-top: 0.5px solid
    ${(props) => props.theme['navbarDropdownContentBorderColor']};
  box-sizing: border-box;

  @media (min-width: ${(props) => props.theme.screenSizes.sm}) {
    flex-grow: 1;
    flex-shrink: 1;
    flex-basis: auto;
    width: auto;
    padding: 0.75rem 1rem;
    border-top: none;

    & > :last-child {
      display: none;
    }

    &:hover {
      & > :last-child {
        display: block;
      }
    }
  }
`

const NavbarNavItem = ({ children, ...rest }) => (
  <Wrapper {...rest}>{children}</Wrapper>
)

NavbarNavItem.propTypes = {
  children: PropTypes.node,
}

export default NavbarNavItem
