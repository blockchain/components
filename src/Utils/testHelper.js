import React from 'react'
import { shallow, mount } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import theme from '../Tools/theme'

function wrapWithTheme(fn, children, options) {
  const wrapper = fn(
    <ThemeProvider theme={theme}>{children}</ThemeProvider>,
    options,
  )

  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext(),
  })
}

export function shallowWithTheme() {
  return wrapWithTheme(shallow, ...arguments)
}

export function mountWithTheme() {
  return wrapWithTheme(mount, ...arguments)
}
