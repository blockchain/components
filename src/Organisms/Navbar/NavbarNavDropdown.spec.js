import React from 'react'
import { mountWithTheme } from '../../Utils/testHelper'

import NavbarNavDropdown from './NavbarNavDropdown'

describe('NavbarNavDropdown', () => {
  it('should render correctly', () => {
    const tree = mountWithTheme(<NavbarNavDropdown />)

    expect(tree.find(NavbarNavDropdown)).toMatchSnapshot()
  })
})
