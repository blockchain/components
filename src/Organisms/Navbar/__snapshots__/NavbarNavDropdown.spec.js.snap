// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarNavDropdown should render correctly 1`] = `
.c0 {
  position: relative;
  width: 100%;
  padding: 0.5rem 0;
  border-top: 0.5px solid #999B9E;
  box-sizing: border-box;
}

@media (min-width:48rem) {
  .c0 {
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
    -webkit-flex-shrink: 1;
    -ms-flex-negative: 1;
    flex-shrink: 1;
    -webkit-flex-basis: auto;
    -ms-flex-preferred-size: auto;
    flex-basis: auto;
    width: auto;
    padding: 0.75rem 1rem;
    border-top: none;
  }

  .c0 > :last-child {
    display: none;
  }

  .c0:hover > :last-child {
    display: block;
  }
}

<NavbarNavItem>
  <styled.li>
    <li
      className="c0"
    />
  </styled.li>
</NavbarNavItem>
`;
